### PR TITLE
Ritesh tab previews outline switching 

### PR DESCRIPTION
**Pull request by riteshsb**
Got tab previews AI switching to work with dummy images! There is a blue outline box as you hold command and multiple press Q which to indicate selecting a tab. There are some issues to be resolved though:

1. I don't know if its correctly aligning the right last switched tab with each image as its hard to tell with dummy images. It is janky and does not switch to a certain number tab sometimes. There are some null errors showing up on the extension debugger on developer mode. 
2. The UI is also janky as the white textbox randomly appears outside the images for some pages.
3. Lastly once we do use the data urls from the actual tabs I want to check the aspect ratio of each image. Right now I made it 200x200 to fit the screen for 5 images but we have to see if the HD full display and aspect ratio will still remain. We can use 'auto' for that to automatically calculate the height or width

### CODE REVIEW

**Review by sileneyang1226 — APPROVED** [review]

File: src/command-testing.js [hunk 1]
```diff
@@ -7,6 +7,39 @@ var previewId = "SmarterTabSwitchingPreviewPopupBox";
...
...QsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgDQBCwAAAIA0AQsAAACANAELAAAAgLAx/gfQlv+5vGWkyAAAAABJRU5ErkJggg=="
```
Comment (Soreloser2) [inline src/command-testing.js:11]: Does this mean we can delete the `preview-pics` directory?

File: src/command-testing.js [hunk 2]
```diff
@@ -94,23 +127,53 @@ window.addEventListener('keyup', (event) => {
...
+                // var pic2 = createImage(previewPic2, 100, 100, "10px");
+                // var pic3 = createImage(previewPic3, 100, 100, "10px");
+                // var pic4 = createImage(previewPic4, 100, 100, "10px");
+                // var pic5 = createImage(previewPic5, 100, 100, "10px");
+
+                // text_div.appendChild(text);
+                // text_div.appendChild(pic2);
+                // text_div.appendChild(pic3);
+                // text_div.appendChild(pic4);
+                // text_div.appendChild(pic5);
```
Comment (Soreloser2) [inline src/command-testing.js:140]: nit: delete comments

**Review by Soreloser2 — APPROVED** [review]
Works for me! A couple notes, only the first item is really important(ish):
- Add `.DS_Store` to the `.gitignore` file. It's a Mac-specific device file so we don't want it in our repo.
- Doesn't need to be in this PR but we should remove the hardcoded 5 images bit. It can be when we capture the tab preview images since that's when we gather info about the actual queue.
- Tests may help us from feature regressions and help with debugging some of the issues mentioned above. Also. could come in future PR with the actual images.

File: src/command-testing.js (hunk 1 above)
Comment (paulstn) [reply src/command-testing.js:11]: Yes

**Review by paulstn — CHANGES_REQUESTED** [review]
I really like the work here! Its very well done, I just have a few comments.
I tried this out locally and noticed that the first time we press CTRL + Q for multiple switching, the outline doesn't appear over the second box. I think that this only requires some repeated code that you've already written.
Secondly, we've had all of the popup code within a listener. I think its time to take it out and clean up the generation of the popup HTML code. There are also a lot of redundant comments, from old code that doesn't need to be there.

File: src/command-testing.js [hunk 3]
```diff
@@ -94,23 +127,53 @@ window.addEventListener('keyup', (event) => {
...
+
+                // Updates the outline to highlight the active image:
+                // numSwitches: number of times Q is clicked while CNTRL is held down
+                // numImages: number of images in preview box
+                var currentIndex = numSwitches % numImages;
+
+                 // Update the 3 pixel thick blue outline to highlight the active image
+                images.forEach((image, index) => {
+                    if (index === currentIndex) {
+                      image.style.outline = '3px solid blue';
```
Comment (paulstn) [inline src/command-testing.js:171]: @riteshsb I think this is the code that you can just factor out and call again at around line 146, to make sure the outline appears around the second box upon initialization.

**Pull request merged**